ConnectivityDemos/ConnectivityDemos/.settings/org.eclipse.jdt.core.prefs

```diff
@@ -6,3 +6,5 @@ org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
 org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
 org.eclipse.jdt.core.compiler.release=enabled
 org.eclipse.jdt.core.compiler.source=11
+org.eclipse.jdt.core.compiler.problem.unusedImport=ignore

```

Comment:
actually I liked the old setting much better: I do not want to ship demos with unused imports...

Reply:
Some unused imports come from javaDoc links i.e. see [SignalInvoker.java](https://github.com/ivy-samples/ivy-project-demos/blob/master/ConnectivityDemos/ConnectivityDemos/src/com/axonivy/connectivity/rest/aynch/SignalInvoker.java) on line 16 we link to a class which we now have to import. The compiler now thinks it's unused but it's not. --> I just realized we could replace those with the fully qualified class path to get rid of the import, I will fix those

Then there are unused import warnings from the swagger petstore example which as much as I understand generates classes from a json and just adds every found type as an import to every class even though it is not necessary in each class --> maybe we could do something here but then I would have to sink time into a story unrelated feature

Reply:
yeah javadocs are akward, but cool that you already found the cause behind.

regarding swagger petstore: these warnings can be ommitted in the IDE by setting the 'ignore optional compiler error' options. 

...so we should not see such warnings if ConnectiviyDemos project is correctly configured. At least in the real Designer.ide 
... if there are now warnings that only occur in the ci-plugin / but no in the real dev environment we should search a better fix for that. Simply turning these warnings of for both environments is not a solution for me.

Reply:
![jdtIgnoreCompileErrs](https://user-images.githubusercontent.com/15085693/75525651-81b01a80-5a10-11ea-9590-8bfab5cc314f.png)

Reply:
I did some digging and found: https://bugs.eclipse.org/bugs/show_bug.cgi?id=506781

That "Ignore optional compile problems" option is written to the .classpath file and is not considered by the compiler.
But now that we have the varArgs parameter :) we can pass a folder (i.e. the src_generated/io) and tell the compiler to ignore all warnings from this folder

The downside is that we currently have to use the nightly engine for building (not yet built on the nightly)

Reply:
if a var args can handle that for now - it's ok. 
if its a frequent real world issue we could interpret the classpath flag on our own .... we do heavily rely upon it anyway.